mucommander-core/src/main/java/com/mucommander/ui/terminal/TerminalIntegration.java

```diff
@@ -0,0 +1,304 @@
+        this.mainFrame = mainFrame;
+        this.verticalSplitPane = verticalSplitPane;
+        prepareVerticalSplitPaneForTerminal();
+    }
+
+    /**
+     * Toggles the Terminal, i.e. shows (maximized) or hides it (minimized).
+     */
+    public void toggleTerminal() {
+        if (!isTerminalMaximized() || terminal == null) {
+            showTerminal();
+        } else {
+            hideTerminal();
+        }
+    }
+
+    private void setTerminalMaximized(boolean terminalMaximized) {
+        this.terminalMaximized = terminalMaximized;
+    }
+
+    private boolean isTerminalMaximized() {
+        return terminalMaximized;
+    }
+
+    private KeyAdapter termCloseKeyHandler() {
+        return new KeyAdapter() {
+            public void keyPressed(KeyEvent keyEvent) {
+                KeyStroke pressedKeyStroke = KeyStroke.getKeyStrokeForEvent(keyEvent);
+                KeyStroke accelerator = ActionKeymap.getAccelerator(
+                        ActionId.asTerminalAction(ActionType.ToggleTerminal.getId()));
+                KeyStroke alternateAccelerator = ActionKeymap.getAlternateAccelerator(
+                        ActionId.asTerminalAction(ActionType.ToggleTerminal.getId()));
+                if (pressedKeyStroke.equals(accelerator) || pressedKeyStroke.equals(alternateAccelerator)) {
+                    keyEvent.consume();
+                    hideTerminal();
+                } else if (!terminal.getTtyConnector().isConnected()) {
+                    // just close terminal if it is not active/connected (for example when sb typed 'exit')
+                    hideTerminal();
+                }
+            }
+        };
+    }
+
+    private JediTermWidget getTerminal(String initialPath) {
+        return TerminalWidget.createTerminal(initialPath);
+    }
+
+    private void showTerminal() {
+        try {
+            LOGGER.info("Going to show Terminal...");
+            String newCwd = mainFrame.getActivePanel().getCurrentFolder().getAbsolutePath();
+            // If !connected means that terminal process has ended (via `exit` command for ex.).
+            if (terminal == null || !terminal.getTtyConnector().isConnected()) {
+                Cursor orgCursor = mainFrame.getCursor();
+                try {
+                    mainFrame.setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
+                    terminal = getTerminal(newCwd);
+                    terminal.getTerminalPanel().addFocusListener(new FocusAdapter() {
+                        public void focusGained(FocusEvent e) {
+                            syncCWD(mainFrame.getActivePanel().getCurrentFolder().getAbsolutePath());
+                        }
+                    });
+                    cwd = newCwd;
+                    terminal.getTerminalPanel().addCustomKeyListener(termCloseKeyHandler());
+                    // TODO do this better? For now 2 lines ~height + 20%
```

Comment:
TODO for myself - comment after I play with that

Reply:
after playing with it, the height looks good to me
but I'd suggest to not to minimize/hide the terminal when mucommander starts, it would be better to show it, in its minimal size, so users would notice it
I also miss the option to switch back from forth from the terminal to the file table using the keyboard - there's no shortcut for this, right?

Reply:
I think that since Terminal uses resources, I'm not sure if it is really welcomed change? We may add an options to Preferences to control that?

Reply:
> I also miss the option to switch back from forth from the terminal to the file table using the keyboard - there's no shortcut for this, right?

You mean when both are visible you would like to switch active panels? (Otherwise F12)

Reply:
no, not to switch active panels but rather something like the following scenario:
1. switch to the terminal
2. make the terminal a big larger
3. start building something (e.g., `mvn install`)
4. switch back to the folder table and navigate to a sub folder there while keep seeing the terminal and the progress of the build in the terminal below

Reply:
this is what I meant. For that there's no shortcut - we may think about it outside of this PR I think

Reply:
> I think that since Terminal uses resources, I'm not sure if it is really welcomed change? We may add an option to Preferences to control that?

And I think there are two types of users - geeks that are familiar with terminal and civilians that might be scared by "black window" - those who come from Windows' Total Commander. So, for "geeks" I would just add preferences panel to launch terminal in minimized 2-liner form (by default set to `false`).  Tell me what you think and I will add it.

Reply:
> this is what I meant. For that there's no shortcut - we may think about it outside of this PR I think

ah ok, I thought you meant the two file table panels, didn't realize you also referred to the terminal as a panel

> And I think there are two types of users - geeks that are familiar with terminal and civilians that might be scared by "black window" - those who came from Windows' Total Commander. So, for "geeks" I would just add preferences panel to launch terminal in minimized 2-liner form (by default set to false). Tell me what you think and I will add it.

I think it makes sense to make configurable, I'm not sure if disabling it by default make more sense than enabling it by default - if we hide the terminal by default, it's less visible for users that may be interested in using an integrated terminal. how about showing the terminal by default and add an option, like `customize command bar`, that can be used to set that preference you proposed to `false`?

Reply:
OK, however `customize command bar` even for me is kinda ambiguous, we need a better wording for that

Reply:
sure, we can find better wording
this PR introduce an improvement even without this part, would you like to include it here or do it separately?

Reply:
> sure, we can find better wording this PR introduce an improvement even without this part, would you like to include it here or do it separately?

I will provide a separate PR for that. In the meantime, I've rebased this PR and added feature description to readme.